### PR TITLE
feat: add field accessors for mailbuilder

### PR DESCRIPTION
**Pull request by andylibrian**
Hi, this PR is to implement the field accessors for `MailBuilder` as proposed in #260

### CODE REVIEW

**Comment by jhillyerd** [comment]
I need to take a deeper look, but my initial concern with the values being returned is that they may allow the original value to be mutated -- which goes against the expectations of a Go builder.  I think we'll probably need to do some copying or deep copying on the returned values.

**Comment by andylibrian** [comment]
Hi @jhillyerd, thanks for letting me know your concern. I'm ok with returning the copied values. Once you are sure this is the way to go, please let me know, I'll update the PR to use returned values instead.

File: builder.go [hunk 1]
```diff
@@ -73,6 +88,11 @@ func (p MailBuilder) ToAddrs(to []mail.Address) MailBuilder {
 	return p
 }
 
+// GetTo returns the stored to addresses.
+func (p *MailBuilder) GetTo() []mail.Address {
+	return p.to
```
Comment (jhillyerd) [inline builder.go:93]: Let's use the following style to return a copy of the to slice:

var dst []string
dst = append(dst, src...)

File: builder.go [hunk 2]
```diff
@@ -88,6 +108,11 @@ func (p MailBuilder) CCAddrs(cc []mail.Address) MailBuilder {
 	return p
 }
 
+// GetCC returns the stored cc addresses.
+func (p *MailBuilder) GetCC() []mail.Address {
+	return p.cc
```
Comment (jhillyerd) [inline builder.go:113]: ditto for this one

File: builder.go [hunk 3]
```diff
@@ -106,13 +131,23 @@ func (p MailBuilder) BCCAddrs(bcc []mail.Address) MailBuilder {
 	return p
 }
 
+// GetBCC returns the stored bcc addresses.
+func (p *MailBuilder) GetBCC() []mail.Address {
+	return p.bcc
```
Comment (jhillyerd) [inline builder.go:136]: and this one

File: builder.go [hunk 4]
```diff
@@ -125,18 +160,33 @@ func (p MailBuilder) Header(name, value string) MailBuilder {
...
+
 // Text returns a copy of MailBuilder that will use the provided bytes for its text/plain Part.
 func (p MailBuilder) Text(body []byte) MailBuilder {
 	p.text = body
 	return p
 }
 
+// GetText returns the stored text/plain part.
+func (p *MailBuilder) GetText() []byte {
+	return p.text
```
Comment (jhillyerd) [inline builder.go:176]: this will also need to be copied.

File: builder.go [hunk 5]
```diff
@@ -125,18 +160,33 @@ func (p MailBuilder) Header(name, value string) MailBuilder {
...
+
 // HTML returns a copy of MailBuilder that will use the provided bytes for its text/html Part.
 func (p MailBuilder) HTML(body []byte) MailBuilder {
 	p.html = body
 	return p
 }
 
+// GetHTML returns the stored text/html part.
+func (p *MailBuilder) GetHTML() []byte {
+	return p.html
```
Comment (jhillyerd) [inline builder.go:187]: and this one.

**Review by jhillyerd — CHANGES_REQUESTED** [review]
Thanks, looks like we only need to worry about mutability with the slice types.

**Review by jhillyerd — APPROVED** [review]

**Comment by jhillyerd** [comment]
Thanks!

**Pull request merged**